model_training/get_current_lr.py

```diff
@@ -0,0 +1,13 @@
+#################################################################
+# --------->>>| Function that always returns current learning
+# --------->>>| rate even for different schedulers
+# ###############################################################
+def current_lr(optim, sched=None):
+    # Varlid for modern schedulers: StepLR, CosineAnnealingLR, etc.
+    if sched is not None and hasattr(sched, "get_last_lr"):
+        try:
+            return sched.get_last_lr()[0]
+        except Exception:
+            pass
```

Comment:
If this fails the code should quit, not fail silently. I would prefer a single line instead of a try-except block.